### ISSUE TITLE
Issues with sideloaded permissions in group

### ISSUE THREAD

**Issue by aleemb** [cartalyst/sentry#539]
I am using the very handy `setPermissionsAttribute()` on the Group and User models (Eloquent) for loading permissions. This works find, however, I am having to do the following:

```
$groups = Sentry::findAllGroups();
foreach ($groups as $group) {
    if ($group->name == 'foo') {
        $group->setPermissionsAttribute('foo.bar');
    }
}

$user = Sentry::getUser();
$user->inGroup('foo'); // true, user is in this group
$user->hasPermission('foo.bar'); // this should be true but it's false
```

To work around this, I am having to do:
```
foreach ($user->getGroups as $userGroup) {
    if ($group->name == 'foo') {
        $group->setPermissionsAttribute('foo.bar');
    }
}
```

...which is problematic when dealing with multiple users.

**Comment by brunogaspar** [comment]
Can you elaborate on your issue? I'm very confused on what you're expecting to happen here.

**Comment by aleemb** [comment]
Sure,

`$user->hasPermission('foo.bar');` should return `true` not `false`. This is because `$user` in the group `foo`, and for that group I already called `$group->setPermissionsAttribute('foo.bar');`

**Comment by aleemb** [comment]
Essentially, if I load permissions directly on `$user` object via `$user->setPermissionsAttribute()` things work fine.

However, when I load a permission `'foo.bar'` directly on the `$group` object via `$group->setPermissionsAttribute()` and if a user belongs to that group, then I expect the `$user->getMergedPermissions()` to contain `'foo.bar'` but it doesn't.

**Comment by suwardany** [comment]
@aleemb you should store the user/group object after setting the permissions so that it can be fetched from the database.

You also don't have to call the `setPermissionsAttribute` method here, you simply do

```
$group->permissions = ['foo.bar' => 1];
$group->save();
```

That way, the permissions are fetched from the database on subsequent calls.

**Comment by aleemb** [comment]
@suwardany, yes that's exactly right but for my use-case I want to sideload the permissions and bypass the DB. In fact, in my app the DB contains null permissions and all permissions are loaded at runtime.
